lib/block_view.go

```diff
@@ -7349,53 +7688,73 @@ func (bav *UtxoView) _connectAcceptNFTBid(
 	nftPostEntry.NumNFTCopiesForSale--
 	bav._setPostEntryMappings(nftPostEntry)
 
-	// Add an operation to the list at the end indicating we've connected an NFT bid.
-	utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
-		Type:                      OperationTypeAcceptNFTBid,
-		PrevNFTEntry:              prevNFTEntry,
-		PrevPostEntry:             prevPostEntry,
-		PrevCoinEntry:             &prevCoinEntry,
-		DeletedNFTBidEntries:      deletedBidEntries,
-		NFTPaymentUtxoKeys:        nftPaymentUtxoKeys,
-		NFTSpentUtxoEntries:       spentUtxoEntries,
-		PrevAcceptedNFTBidEntries: prevAcceptedBidHistory,
-
-		// Rosetta fields.
-		AcceptNFTBidCreatorPublicKey: nftPostEntry.PosterPublicKey,
-		AcceptNFTBidBidderPublicKey: bidderPublicKey,
-		AcceptNFTBidCreatorRoyaltyNanos: creatorCoinRoyaltyNanos,
-	})
+	if txMeta.IsAcceptNFTBid {
+		// Add an operation to the list at the end indicating we've connected an NFT bid.
+		utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+			Type:                      OperationTypeAcceptNFTBid,
+			PrevNFTEntry:              prevNFTEntry,
+			PrevPostEntry:             prevPostEntry,
+			PrevCoinEntry:             &prevCoinEntry,
+			DeletedNFTBidEntries:      deletedBidEntries,
+			NFTPaymentUtxoKeys:        nftPaymentUtxoKeys,
+			NFTSpentUtxoEntries:       spentUtxoEntries,
+			PrevAcceptedNFTBidEntries: prevAcceptedBidHistory,
+			// Rosetta fields.
+			AcceptNFTBidCreatorPublicKey: nftPostEntry.PosterPublicKey,
+			AcceptNFTBidBidderPublicKey: bidderPublicKey,
+			AcceptNFTBidCreatorRoyaltyNanos: creatorCoinRoyaltyNanos,
+		})
+	}
+
+	if txMeta.IsNFTBid {
+		// Add an operation to the list at the end indicating we've connected an NFT bid.
+		utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
```

Comment:
It seems error prone to duplicate so much code here.  We should build the part of the UtxoOperation that is consistent regardless of IsAcceptNFTBid vs. IsNFTBid and then selectively add the Rosetta fields with if/else if.

Reply:
You also might want to sanity check that IsAcceptNFTBid and IsNFTBid are never true at the same time.